test/parallel/sdk.API.claimRewards.test.ts

```diff
@@ -0,0 +1,450 @@
+/*
+ *
+ * @group sdk
+ */
+import { jest } from "@jest/globals";
+import { Keyring } from "@polkadot/api";
+import { getApi, initApi } from "../../utils/api";
+import { Assets } from "../../utils/Assets";
+import { BN } from "@polkadot/util";
+import { setupApi, setupUsers } from "../../utils/setup";
+import { Sudo } from "../../utils/sudo";
+import { User } from "../../utils/User";
+import { getEnvironmentRequiredVars } from "../../utils/utils";
+import { Xyk } from "../../utils/xyk";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import {
+  claimRewardsAll,
+  getLiquidityAssetId,
+  getRewardsInfo,
+  promotePool,
+} from "../../utils/tx";
+import {
+  getEventResultFromMangataTx,
+  getNextAssetId,
+} from "../../utils/txHandler";
+import { ExtrinsicResult, waitForRewards } from "../../utils/eventListeners";
+import { BN_ZERO } from "@mangata-finance/sdk";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.setTimeout(2500000);
+process.env.NODE_ENV = "test";
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+let testUser: User;
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let soloTokenId: BN;
+let poolTokenIds: BN[];
+let liqIds: BN[];
+let tokenAmounts: BN[];
+const defaultCurrencyValue = new BN(250000);
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  keyring = new Keyring({ type: "sr25519" });
+
+  // setup users
+  sudo = new User(keyring, sudoUserName);
+
+  await setupApi();
+  await setupUsers();
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.FinalizeTge(),
+    Assets.initIssuance(),
+    Assets.mintNative(sudo)
+  );
+
+  [testUser] = setupUsers();
+
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+
+  [soloTokenId] = await Assets.setupUserWithCurrencies(
+    sudo,
+    [defaultCurrencyValue],
+    sudo
+  );
+
+  const batchPromisesMinting = [];
+  const batchPromisesPromoting = [];
+  poolTokenIds = [];
+  tokenAmounts = [];
+  liqIds = [];
+
+  const nextTokenId = (await getNextAssetId()).toNumber();
+
+  for (
+    let newTokenId = nextTokenId;
+    newTokenId < nextTokenId + 13;
+    newTokenId++
+  ) {
+    poolTokenIds.push(new BN(newTokenId));
+
+    tokenAmounts.push(defaultCurrencyValue);
+
+    batchPromisesMinting.push(
+      Assets.mintToken(new BN(newTokenId), testUser, defaultCurrencyValue)
```

Comment:
if we want those tests run in parallel, we may need to find some other way to do this.

Reply:
I am not sure that I understand the issue. We can create another test file for checking users with 10 pools and over, but in any case, we need to create these pools...

Reply:
No, I mean that creating tokens in a loop, wont work on parallel scenarios. for example: 
This loop runs, and consider that his next assetId is 12. 
but in the meantime, another test is running, so the 12 is already generated for another test. 
This test will continue, but actions like creating pools and such will fail because the token will collide with some other test.  ( or event worst, the other test fails because of the action this test does. )

Reply:
Ahh, you're right! Changed

Reply:
okk, lets see how it goes, perhaps we should improve the setup of the tokens somehow